Model/Data/AmwalOrderDetails.php

```diff
@@ -0,0 +1,116 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Amwal\Payments\Model\Data;
+
+use Amwal\Payments\Api\Data\AmwalOrderInterface;
+use Magento\Framework\Webapi\Rest\Request;
+use Magento\Sales\Api\OrderRepositoryInterface;
+use Magento\Framework\Api\SearchCriteriaBuilder;
+use Magento\Store\Model\StoreManagerInterface;
+
+class AmwalOrderDetails implements AmwalOrderInterface
+{
+    protected $orderRepository;
+    protected $searchCriteriaBuilder;
+    private Request $restRequest;
+    private StoreManagerInterface $storeManager;
+
+    public function __construct(OrderRepositoryInterface $orderRepository, SearchCriteriaBuilder $searchCriteriaBuilder, Request $restRequest, StoreManagerInterface $storeManager)
+    {
+        $this->orderRepository = $orderRepository;
+        $this->searchCriteriaBuilder = $searchCriteriaBuilder;
+        $this->restRequest = $restRequest;
+        $this->storeManager = $storeManager;
+    }
+
+    public function getOrderDetails($amwalOrderId)
+    {
+        // Get order by Amwal order ID
+        $order = $this->getOrderByAmwalOrderId($amwalOrderId);
+        $order->setData('order_url', $this->getOrderUrl($order));
+
+        return [
+            'order' => $order->getData(),
+        ];
+    }
+
+    public function updateOrderStatus()
+    {
+        $requestBody = $this->restRequest->getBodyParams();
+
+        $amwalOrderId = $requestBody['amwal_order_id'];
+        $orderId      = $requestBody['order_id'];
+        $refId        = $requestBody['ref_id'];
+        $status       = $requestBody['status'];
+
+        // Validate the provided status
+        if (!$this->isValidStatus($status)) {
+            return false;
+        }
+        try {
+            $order = $this->getOrderByAmwalOrderId($amwalOrderId, $orderId, $refId);
+
+            // Update order status
+            $order->setStatus($status);
```

Comment:
You need to validate if the status is within valid status like ( pending, canceled, failed ) otherwise we are not editing the status directly. For example if the status in Magento is completed and we call this endpoint, the status will return back to processing

Second thing you do not have to pass the status in the request, you just check if the status in Magento is valid status ( pending, canceled, failed )  then you change it to processing.

Reply:
I'm already check the status is valid or not, but we will remove it and set the order status in PayOrder logic